src/test/codegen/enum-repeat.rs

```diff
@@ -0,0 +1,19 @@
+// compile-flags: -O
+
+#![crate_type = "lib"]
+
+// CHECK-LABEL: @none_repeat
+#[no_mangle]
+pub fn none_repeat() -> [Option<u8>; 64] {
+    // CHECK: store <128 x i8>
+    // CHECK-NEXT: ret void
+    [None; 64]
+}
+
+// CHECK-LABEL: @some_repeat
+#[no_mangle]
+pub fn some_repeat() -> [Option<u8>; 64] {
+    // CHECK: store <128 x i8>
+    // CHECK-NEXT: ret void
+    [Some(0); 64]
```

Comment:
It's interesting to me that `some_repeat` seems to do 16 bytes at a time on x64, but `none_repeat` doesn't.  Might be interesting to look at why LLVM is treating them differently.

Also, out of curiosity, what's the assembly difference before/after this PR?

Reply:
[godbolt](https://godbolt.org/z/68xrb14EG) - This is `none_repeat` before and after LLVM opts - it looks like LLVM is optimising out the undefined write and forgetting about it. This happened even if I explicitly added `store undef`.

The current assembly output is
- `None; 64` - 64 single byte `mov`s to the tag
- `Some(0); 64` - A 16 byte constant vector (alternating 0/1) is stored into the array with `movups`
- `Some(1); 64` - The same as `Some(0)`, but with an all-1's pattern

With this patch
- `None; 64` - 16 byte vector created with `xorps`, then `movups`
- `Some(0); 64`/`Some(1); 64` - Same as before